src/Components/Admins/Form/index.jsx

```diff
@@ -1,26 +1,88 @@
 import { useState, useEffect } from 'react';
+import { Link, useParams, useHistory } from 'react-router-dom';
+import { Input } from 'Components/Shared/Inputs';
 import { useDispatch, useSelector } from 'react-redux';
-import { getAdmins, updateAdmin, addAdmin } from '../../../redux/admins/thunks';
+import { getAdmins, updateAdmin, addAdmin } from 'redux/admins/thunks';
+import { useForm } from 'react-hook-form';
+import { joiResolver } from '@hookform/resolvers/joi';
+import Joi from 'joi';
 import styles from './adminsForm.module.css';
-import { Input } from '../../Shared/Inputs';
-import { Link, useParams, useHistory } from 'react-router-dom';
-import Button from '../../Shared/Button';
-import Modal from '../../Shared/Modal';
-import * as actionsConstants from '../../../redux/admins/actions';
+import Button from 'Components/Shared/Button';
+import Modal from 'Components/Shared/Modal';
+import * as actionsConstants from 'redux/admins/actions';
 
 const Form = () => {
+  const history = useHistory();
+  const { id } = useParams();
+  const dispatch = useDispatch();
   const { isPending, data, error, errorSwitch, message } = useSelector((state) => state.admins);
   const [inputs, setInputs] = useState({});
   const [successModal, setSuccesModal] = useState(false);
   const [errorModal, setErrorModal] = useState(false);
-  const history = useHistory();
-  const dispatch = useDispatch();
-  const { id } = useParams();
+
+  const schema = Joi.object({
+    firstName: Joi.string()
+      .min(3)
+      .max(15)
+      .pattern(/^[a-zA-Z-]+$/)
+      .messages({
+        'string.pattern.base': 'First name must contain only letters'
+      }),
+    lastName: Joi.string()
+      .min(3)
+      .max(15)
+      .pattern(/^[a-zA-Z-]+$/)
+      .messages({
+        'string.pattern.base': 'Last name must contain only letters'
+      }),
+    dni: Joi.string()
+      .pattern(/^[0-9]+$/)
+      .min(7)
+      .max(8)
+      .messages({
+        'string.pattern.base': 'DNI must contain only numbers'
+      }),
+    phone: Joi.number().min(1000000000).max(9999999999),
```

Comment:
When you enter an invalid phone it says "phone" must be greater than or equal to 1000000000, first of all, this message isn't very clear for the user, it has a ver long number hard to read, and the number doesn't have to be greater than that, the correct message is "Phone must have 10 numbers"